openedx/core/djangoapps/content/search/documents.py

```diff
@@ -0,0 +1,215 @@
+"""
+Utilities related to indexing content for search
+"""
+from __future__ import annotations
+import hashlib
+import logging
+
+from django.utils.text import slugify
+from opaque_keys.edx.keys import UsageKey, LearningContextKey
+
+from openedx.core.djangoapps.content_libraries import api as lib_api
+from openedx.core.djangoapps.content_tagging import api as tagging_api
+from openedx.core.djangoapps.xblock import api as xblock_api
+
+log = logging.getLogger(__name__)
+STUDIO_INDEX_NAME = "studio_content"
+
+
+class Fields:
+    """
+    Fields that exist on the documents in our search index
+    """
+    # Meilisearch primary key. String.
+    id = "id"
+    usage_key = "usage_key"
+    type = "type"  # DocType.course_block or DocType.library_block (see below)
```

Comment:
If you want to change any of [the index settings](https://github.com/open-craft/edx-platform/blob/1b5dfdf3f89058e542fdb6de8be1cd8bbe78b074/openedx/core/djangoapps/content/search/management/commands/reindex_studio.py#L62-L79):
* which attributes can be used for filtering/faceted search
* which attributes are used for keyword search
* The fact that the "usage_key" is the distinct attribute

...then it's best to do it via `reindex_studio` management command, which is the only method I've implemented for doing so.

The Meilisearch docs [recommend](https://www.meilisearch.com/docs/reference/api/settings#filterable-attributes) this approach:
> Updating \[index settings] will re-index all documents in the index, which can take some time. We recommend updating your index settings first and then adding documents as this reduces RAM consumption.

**But** otherwise just adding new fields or values can be done any time. If it's something like a new value for the `type` field, or a new field that doesn't need to be used for filtering or keyword search, or it's a dictionary value within one of the existing fields (e.g. a new set of keys and values in the `content` field from some new XBlock's index_dictionary) - then it can be done anytime, no migration needed, no performance issue.

Reply:
Noted via a comment:

https://github.com/openedx/edx-platform/blob/ec0c4a875d7bc4766ff02c50a5e36d8ec6cd3237/openedx/core/djangoapps/content/search/documents.py#L52-L54